main.cpp

```diff
@@ -130,40 +179,58 @@ struct CarWash
 
 struct Cat
 {
-    // number of limbs
     int numLimbs = 3;
-    // number of tails    
     int numTails = 1;
-    // age
     float ageYears = 3.6f;
-    // colour
     std::string colour = "orange";
-    // weight
     float weightKg = 4.1f;
+    int cutenessLevel= 50;
 
-    // 3 things it can do:
-    // meow
-    void meow(float loudnessDB);
+    void meow(int loudnessDB);
 
-    // catch small animals
-    bool catchAnimal(std::string animalSpecies); // returns true if the animal is caught
+    bool catchAnimal(std::string animalSpecies); 
 
-    // throw up fur balls
-    float throwUpFurBall(); // returns the size of the fur ball in mm
+    float throwUpFurBall(); 
 };
 
+void Cat::meow(int loudnessDB)
+{
+    if (loudnessDB < 30)
+    {
+        cutenessLevel += 10;
+    }
+    else
+    {
+        cutenessLevel -= loudnessDB;        
+    }
+}
+
+bool Cat::catchAnimal(std::string animalSpecies)
+{
+    if (animalSpecies == "Fly")
+    {
+        cutenessLevel += 10;
+        return true;
+    }
+    else
+    {
+        cutenessLevel -= 10;
+        return false;
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise